Library/Homebrew/formula_installer.rb

```diff
@@ -589,9 +589,16 @@ def link(keg)
       keg.remove_linked_keg_record
     end
 
+    link_overwrite_backup = []
+
     begin
       keg.link
     rescue Keg::ConflictError => e
+      if formula.link_overwrite?(e.dst) && !link_overwrite_backup.include?(e.dst)
+        e.dst.rename "#{e.dst}.bak"
```

Comment:
I don't think we need to be backing up files. It just creates more manual work for the user to cleanup and it'll happen every time you `brew upgrade glib-networking`.